src/inmanta/data/model.py

```diff
@@ -898,6 +898,44 @@ def has_source(self) -> bool:
 LEGACY_PIP_DEFAULT = PipConfig(use_system_config=True)
 
 
+class Discrepancy(BaseModel):
+    """
+    Records a discrepancy between the state as persisted in the database and
+    the in-memory state in the scheduler.
+
+    :param rid: If set, this discrepancy is specific to this resource.
+    :param field: If set, specifies on which field this discrepancy was detected.
+        If left unset, and a rid is specified, the discrepancy was detected on the
+        resource level i.e. it is missing from either the db or the scheduler.
```

Comment:
You could also mention the case of `model_version`

Reply:
Updated docstring.